### DISCUSSION TITLE
New "lock object in the editor" feature is counter productive.

### DISCUSSION THREAD

**Discussion by Oxey405** [4ian/GDevelop#3334] [Enhancement ideas (and their technical discussions)]
## Describe the bug
I do think that the older "lock object in the editor" feature was better because prevented to select it accidently.
I think they should at least be another option wich let disallow selection of an object in the editor.

## To Reproduce
1. Lock an object
2. Hover it with your mouse

## Other details
* Works on every system
* 5.123

**Comment by tristanbob** [comment]
I really like the new lock object behavior.  I usually lock all objects in the scene except the ones I am working on. 

With the old method, if I decide I need to move an object that was already placed and locked, I had to search the instance list and try to find the right one.  With a lot of instances of an object, it involved picking each one and hoping it was the instance I wanted to move.

With the new method, I can easily select the instance and unlock it.

**Reply by Oxey405** [reply]
Well I think that it was better before because you could select multiple objects without caring about accidently clicking on your background object

**Reply by tristanbob** [reply]
I think I might understand your problem:  There is no way to select the objects behind a locked object.   

Is that the main problem you experience?

**Reply by 4ian** [reply]
@Oxey405 Try to give a bit more information about your use case, how are your objects organized, maybe a screenshot :)

**Comment by Silver-Streak** [comment]
I wonder if some of the confusion could be solved by adding an option to have the cursor highlight/tint the object that will be selected if you click?

It definitely should be an option and not default, simply because it could drive people not dealing with a lot of layers a bit mad, but I feel like this would help avoid misclicks.

Another option would be adding an "Above/below" modifier for clicks, like holding Z while clicking will select the lowest Z-order object under the cursor, while holding X while clicking will select the highest Z-order object.

**Comment by arthuro555** [comment]
Yes, generally selection of instances that are on top of each other is a pain, and I think that it should be improved. Personally, i think that locking an instance shouldn't prevent selection, but adding a way to prevent selection would be nice, and that there should be a way to select a specific instance when hovering multiple, maybe through scrolling or arrow keys.
